src/TensorNetwork.jl

```diff
@@ -71,7 +94,17 @@ TensorNetwork(tn::TensorNetwork) = tn
 
 Return a shallow copy of a [`TensorNetwork`](@ref).
 """
-Base.copy(tn::TensorNetwork) = TensorNetwork(tensors(tn))
+function Base.copy(tn::TensorNetwork)
+    new_tn = TensorNetwork(tensors(tn), tn.check_index_sizes[])
```

Comment:
with the comment above, you should rather do:
```suggestion
    new_tn = TensorNetwork(tensors(tn))
    set_unsafe_context!(new_tn, tn.check_index_sizes[])
```

Reply:
You can not do that since there is a index check inside the constructor.